app/controllers/settings/profiles_controller.rb

```diff
@@ -16,7 +16,14 @@ def show
   end
 
   def update
-    if UpdateAccountService.new.call(@account, account_params)
+    update_result = begin
+      UpdateAccountService.new.call(@account, account_params)
+    rescue Mastodon::DimensionsValidationError => de
+      @account.errors.add(:avatar, "#{I18n.t('simple_form.labels.defaults.avatar_img_error')}. #{de.message}")
```

Comment:
I suppose this could be put into UpdateAccountService instead

Reply:
The error message doesn't belong into simple_form. I suggest just using de.message without trying to localize anything for now.